providers/azure/resources/cosmosdb.go

```diff
@@ -75,10 +96,92 @@ func (a *mqlAzureSubscriptionCosmosDbService) accounts() ([]interface{}, error)
 					"properties": llx.DictData(properties),
 				})
 			if err != nil {
-				return nil, err
+				return err
 			}
-			res = append(res, mqlCosmosDbAccount)
+			*res = append(*res, mqlCosmosDbAccount)
 		}
 	}
-	return res, nil
+	return nil
+}
+
+func fetchMongoDBClusters(ctx context.Context, runtime *plugin.Runtime, conn *connection.AzureConnection, subId string, res *[]interface{}) error {
+	resClient, err := armresources.NewClient(subId, conn.Token(), &arm.ClientOptions{
+		ClientOptions: conn.ClientOptions(),
+	})
+	if err != nil {
+		return err
+	}
+	filter := "resourceType eq 'Microsoft.DocumentDB/mongoClusters'"
+	pager := resClient.NewListPager(&armresources.ClientListOptions{
+		Filter: &filter,
+	})
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		for _, account := range page.Value {
+			properties, err := convert.JsonToDict(account.Properties)
+			if err != nil {
+				return err
+			}
+
+			mqlResource, err := CreateResource(runtime, "azure.subscription.cosmosDbService.account",
```

Comment:
i am a bit confused here: the function says its fetching cluster, but we're creating cosmosdb accounts. is this a wrong mql resource here?

Reply:
yes usually MondoDB and PostgreSQL are referred as cluster as a logical group, but here in cosmodb context better to stick to the account naming. Updated